dev/devicelab/bin/tasks/hello_world_impeller.dart

```diff
@@ -0,0 +1,81 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async' show Completer, StreamSubscription;
+import 'dart:io' show Directory, Process;
+
+import 'package:flutter_devicelab/framework/devices.dart'
+    show Device, DeviceOperatingSystem, deviceOperatingSystem, devices;
+import 'package:flutter_devicelab/framework/framework.dart' show task;
+import 'package:flutter_devicelab/framework/task_result.dart' show TaskResult;
+import 'package:flutter_devicelab/framework/utils.dart'
+    show dir, flutter, flutterDirectory, inDirectory, startFlutter;
+import 'package:path/path.dart' as path;
+
+Future<TaskResult> run() async {
+  deviceOperatingSystem = DeviceOperatingSystem.android;
+  final Device device = await devices.workingDevice;
+  await device.unlock();
+  final Directory appDir =
+      dir(path.join(flutterDirectory.path, 'examples/hello_world'));
+
+  bool isUsingValidationLayers = false;
+  bool hasValidationErrors = false;
+  int impellerBackendCount = 0;
+  final Completer<void> didReceiveBackendMessage = Completer<void>();
+
+  await inDirectory(appDir, () async {
+    await flutter('packages', options: <String>['get']);
+
+    final StreamSubscription<String> adb = device.logcat.listen(
+      (String data) {
+        if (data.contains('Using the Impeller rendering backend')) {
+          // Sometimes more than one of these will be printed out if there is a
+          // fallback.
+          if (impellerBackendCount == 0) {
```

Comment:
optional nit: You can use `Completer.isCompleted`.

https://api.dart.dev/dev/3.2.0-150.0.dev/dart-async/Completer/isCompleted.html

Reply:
sgtm, done.